bump_dictionary/utils.py

```diff
@@ -0,0 +1,129 @@
+import json
+from copy import deepcopy
+from pathlib import Path
+from typing import Any, Type
+
+import typer
+from jsonschema import Draft202012Validator
+from pydantic import ValidationError
+
+from .logger import log_error, logger
+from .models.legacy_dictionary_model import (
+    CategoricalNeurobagel,
+    ContinuousNeurobagel,
+    IdentifierNeurobagel,
+    Neurobagel,
+    ToolNeurobagel,
+)
+
+
+def load_json(file: Path) -> Any:
+    """Load a JSON file and return its content if file has valid encoding and is valid JSON."""
+    try:
+        with open(file, "r", encoding="utf-8") as f:
+            return json.load(f)
+    except UnicodeDecodeError:
+        log_error(
+            logger,
+            f"Data dictionary must have UTF-8 encoding: {file}. "
+            "[italic]TIP: Need help converting your file? Try a tool like iconv (http://linux.die.net/man/1/iconv) or https://www.freeformatter.com/convert-file-encoding.html.[/italic]",
+        )
+    except json.JSONDecodeError:
+        log_error(
+            logger,
+            f"Data dictionary is not valid JSON: {file}.",
+        )
+
+
+def check_overwrite(file: Path, overwrite: bool):
+    """Exit program gracefully if an output file already exists but --overwrite is not enabled."""
+    if file.exists() and not overwrite:
+        raise typer.Exit(
+            typer.style(
+                f"Output file {file} already exists. Use --overwrite or -f to overwrite.",
+                fg=typer.colors.RED,
+            )
+        )
+
+
+def patch_schema_to_allow_transformation_or_format(schema: dict) -> dict:
+    """
+    Patch a data dictionary schema to allow for either a 'Format' or 'Transformation' key
+    (but not both) for continuous columns.
+    This ensures that all v1 annotation tool data dictionaries pass the legacy schema validation
+    without encoding the deprecated 'Transformation' key in the data dictionary model itself.
+    """
+    patched_schema = deepcopy(schema)
+    continuous_schema = patched_schema["$defs"]["ContinuousNeurobagel"]
+    continuous_schema["properties"]["Transformation"] = deepcopy(
+        continuous_schema["properties"]["Format"]
+    )
+    if "Format" in continuous_schema.get("required", []):
+        continuous_schema["required"].remove("Format")
+    continuous_schema["oneOf"] = [
+        {"required": ["Transformation"], "not": {"required": ["Format"]}},
+        {"required": ["Format"], "not": {"required": ["Transformation"]}},
+    ]
+    return patched_schema
```

Comment:
Hhm. Why don't we allow "Transformation" in the actual pydantic schema instead. Can have a big fat warning comment above - but since the pydantic class is a copy of the actual schema anyway (and not e.g. a submodule or such), making changes there seems a lot more intuitive than hard-coding this complex logic